h5py/_conv.pyx

```diff
@@ -762,22 +762,22 @@ cdef int conv_ndarray2vlen(void* ipt,
         vlen_t* in_vlen = <vlen_t*>opt
         void* data
         cnp.ndarray ndarray
-        size_t len
+        size_t len, nbytes
         PyObject* buf_obj0
 
     buf_obj0 = buf_obj[0]
     ndarray = <cnp.ndarray> buf_obj0
     len = ndarray.shape[0]
+    nbytes = len * max(outtype.get_size(), intype.get_size())
 
-    if outtype.get_size() > intype.get_size():
-        data = emalloc(outtype.get_size() * len)
-    else:
-        data = emalloc(intype.get_size() * len)
-    memcpy(data, ndarray.data, intype.get_size() * len)
-    H5Tconvert(intype.id, outtype.id, len, data, NULL, H5P_DEFAULT)
+    #with nogil:
+    if 1:
+        data = emalloc(nbytes)
+        memcpy(data, ndarray.data, nbytes)
```

Comment:
I think if `outtype.get_size() > intype.get_size()`, this could read past the end of `ndarray.data`.

Reply:
this is why we took the max of the two a couple of lines before

Reply:
Right, you allocate a buffer to the max of the two. But if the output data is larger than the input data, then copying the input data needs to use the input data size, not the output data size.

Reply:
right. corrected